### PR TITLE
feat(ucs): add new resource policy

### PR DESCRIPTION
**Pull request by Jason-Zhang9309**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/ucs' TESTARGS='-run TestAccPolicy_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/ucs -v -run TestAccPolicy_basic -timeout 360m -parallel 4
=== RUN   TestAccPolicy_basic
=== PAUSE TestAccPolicy_basic
=== CONT  TestAccPolicy_basic
--- PASS: TestAccPolicy_basic (18.37s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/ucs       18.418s
```

### CODE REVIEW

File: docs/resources/ucs_policy.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+Manages a UCS policy resource within HuaweiCloud.
+
+## Example Usage
+
+### Basic Usage
+
+```hcl
+variable "user_id_1" {}
+variable "user_id_1" {}
```
Comment (Lance52259) [inline docs/resources/ucs_policy.md]:
```suggestion
variable "user_id_1" {}
variable "user_id_2" {}
```
or
```
variable "user_ids" {
  type = list(string)
}
```

File: docs/resources/ucs_policy.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+  type         = "admin"
+  description  = "created by terraform"
+}
+```
+
+### Custom Type Policy
+
+```hcl
+variable "user_id_1" {}
+variable "user_id_1" {}
```
Comment (Lance52259) [inline docs/resources/ucs_policy.md]: ditto

File: docs/resources/ucs_policy.md [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+* `region` - (Optional, String, ForceNew) Specifies the region in which to create the resource.
+  If omitted, the provider-level region will be used. Changing this parameter will create a new resource.
+
+* `name` - (Required, String, ForceNew) Specifies the name of the UCS policy. The name consists of 1 to 63 characters,
+  including lowercase letters, digits and hyphens (-), must start with a letter and end with a letter or digit.
+
+  Changing this parameter will create a new resource.
+
+* `iam_user_ids` - (Required, List) Specifies the list of iam user IDs to associate to the policy.
```
Comment (Lance52259) [inline docs/resources/ucs_policy.md]:
```suggestion
* `iam_user_ids` - (Required, List) Specifies the list of IAM user IDs to associate to the policy.
```

File: huaweicloud/config/endpoints.go [hunk 4]
```diff
@@ -158,10 +160,11 @@ var allServiceCatalog = map[string]ServiceCatalog{
...
 	},
-	"aom": {
-		Name:    "aom",
-		Version: "svcstg/icmgr/v1",
-		Product: "AOM",
+	"swr": {
+		Name:             "swr-api",
+		Version:          "v2",
+		WithOutProjectID: true,
+		Product:          "SWR",
```
Comment (Lance52259) [inline huaweicloud/config/endpoints.go:167]: It seems that this modification does not belong to this commit.

File: huaweicloud/services/ucs/resource_huaweicloud_ucs_policy.go [hunk 5]
```diff
@@ -0,0 +1,354 @@
...
+			},
+			"type": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: `Specifies the type of the UCS policy.`,
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
```
Comment (Lance52259) [inline huaweicloud/services/ucs/resource_huaweicloud_ucs_policy.go]:
```suggestion
```
This parameter is support left blank via update function during API .

File: huaweicloud/services/ucs/resource_huaweicloud_ucs_policy.go [hunk 6]
```diff
@@ -0,0 +1,354 @@
...
+func buildPolicyMetadataBodyParams(d *schema.ResourceData) map[string]interface{} {
+	bodyParams := map[string]interface{}{
+		"name": utils.ValueIngoreEmpty(d.Get("name")),
+	}
+	return bodyParams
+}
+
+func buildPolicySpecBodyParams(d *schema.ResourceData) map[string]interface{} {
+	bodyParams := map[string]interface{}{
+		"description": utils.ValueIngoreEmpty(d.Get("description")),
```
Comment (Lance52259) [inline huaweicloud/services/ucs/resource_huaweicloud_ucs_policy.go]:
```suggestion
		"description": d.Get("description"),
```
ditto

File: docs/resources/ucs_policy.md (hunk 1 above)
Comment (Jason-Zhang9309) [reply docs/resources/ucs_policy.md]: fixed

File: docs/resources/ucs_policy.md (hunk 2 above)
Comment (Jason-Zhang9309) [reply docs/resources/ucs_policy.md]: fixed

File: docs/resources/ucs_policy.md (hunk 3 above)
Comment (Jason-Zhang9309) [reply docs/resources/ucs_policy.md]: fixed

File: docs/resources/ucs_policy.md [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+
+* `description` - (Optional, String) Specifies the description of the UCS policy.
+  The description consists of 0 to 255 characters.
+
+* `details` - (Optional, List) Specifies the details of the UCS policy.
+  This only works when the type is **custom**.
+  The [Details](#Policy_Details) structure is documented below.
+
+<a name="Policy_Details"></a>
+The `Content` block supports:
```
Comment (Lance52259) [inline docs/resources/ucs_policy.md]:
```suggestion
The `details` block supports:
```

File: docs/resources/ucs_policy.md [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+
+* `id` - The resource ID.
+
+* `created_at` - The created time.
+
+* `updated_at` - The updated time.
+
+## Import
+
+The health check can be imported using the `id`, e.g.
```
Comment (houpeng80) [inline docs/resources/ucs_policy.md]:
```suggestion
The UCS policy can be imported using the `id`, e.g.
```

File: docs/resources/ucs_policy.md (hunk 7 above)
Comment (Jason-Zhang9309) [reply docs/resources/ucs_policy.md]: fixed

File: docs/resources/ucs_policy.md (hunk 8 above)
Comment (Jason-Zhang9309) [reply docs/resources/ucs_policy.md]: fixed

File: huaweicloud/services/ucs/resource_huaweicloud_ucs_policy.go (hunk 5 above)
Comment (Jason-Zhang9309) [reply huaweicloud/services/ucs/resource_huaweicloud_ucs_policy.go]: fixed

File: huaweicloud/services/ucs/resource_huaweicloud_ucs_policy.go (hunk 6 above)
Comment (Jason-Zhang9309) [reply huaweicloud/services/ucs/resource_huaweicloud_ucs_policy.go]: fixed

File: huaweicloud/config/endpoints.go (hunk 4 above)
Comment (Jason-Zhang9309) [reply huaweicloud/config/endpoints.go:167]: only adjust the order, I think it's ok

**Comment by Jason-Zhang9309** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/ucs' TESTARGS='-run TestAccPolicy_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/ucs -v -run TestAccPolicy_basic -timeout 360m -parallel 4
=== RUN   TestAccPolicy_basic
=== PAUSE TestAccPolicy_basic
=== CONT  TestAccPolicy_basic
--- PASS: TestAccPolicy_basic (18.13s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/ucs       18.175s
```

File: huaweicloud/config/endpoints.go [hunk 9]
```diff
@@ -238,11 +249,6 @@ var allServiceCatalog = map[string]ServiceCatalog{
 		Version: "v2",
 		Product: "VBS",
 	},
-	"sdrs": {
```
Comment (ShiChangkuo) [inline huaweicloud/config/endpoints.go]: we should keep the endpoint of **sdrs**

File: huaweicloud/services/ucs/resource_huaweicloud_ucs_policy.go [hunk 10]
```diff
@@ -0,0 +1,353 @@
...
+func resourcePolicyCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	// createPolicy: Create a UCS Policy.
+	var (
+		createPolicyHttpUrl = "v1/permissions/rules"
+		createPolicyProduct = "ucs"
+	)
+	createPolicyClient, err := cfg.NewServiceClient(createPolicyProduct, region)
```
Comment (ShiChangkuo) [inline huaweicloud/services/ucs/resource_huaweicloud_ucs_policy.go]:
```suggestion
	createPolicyClient, err := cfg.NewServiceClient(createPolicyProduct, "")
```

as the UCS is golbal, we can remove the `region` parameter and create the client with ""

File: huaweicloud/config/endpoints.go (hunk 9 above)
Comment (Jason-Zhang9309) [reply huaweicloud/config/endpoints.go]: sure

File: huaweicloud/services/ucs/resource_huaweicloud_ucs_policy.go (hunk 10 above)
Comment (Jason-Zhang9309) [reply huaweicloud/services/ucs/resource_huaweicloud_ucs_policy.go]: fixed

**Comment by Jason-Zhang9309** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/ucs' TESTARGS='-run TestAccPolicy_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/ucs -v -run TestAccPolicy_basic -timeout 360m -parallel 4
=== RUN   TestAccPolicy_basic
=== PAUSE TestAccPolicy_basic
=== CONT  TestAccPolicy_basic
--- PASS: TestAccPolicy_basic (20.05s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/ucs       20.104s
```
